test/autorest/generated/lrogroup/pollers.go

```diff
@@ -23,25 +25,71 @@ type lrOSCustomHeaderPost202Retry200Poller struct {
 	// the client for making the request
 	client *lrOSCustomHeaderOperations
 	// polling tracker
-	// pt *pollingTrackerPost
+	pt *pollingTrackerPost
 }
 
+// Done returns true if the polling operation has terminated either in a success case or failure case,
+// otherwise it will return false
 func (p *lrOSCustomHeaderPost202Retry200Poller) Done() bool {
-	return false
+	return p.pt.hasTerminated()
 }
 
+// ID ...
 func (p *lrOSCustomHeaderPost202Retry200Poller) ID() string {
-	return "NYI"
+	return ""
 }
 
 func (p *lrOSCustomHeaderPost202Retry200Poller) Poll(ctx context.Context) (*LrOSCustomHeaderPost202Retry200Response, error) {
-	return nil, nil
+	done, err := p.doneWithContext(ctx)
+	if done {
+		resp := p.response()
+		result, err := p.client.post202Retry200HandleResponse(resp)
+		if err != nil {
+			return nil, err
+		}
+		return result, nil
+	}
+	return nil, err
 }
 
 func (p *lrOSCustomHeaderPost202Retry200Poller) Wait(ctx context.Context, pollingInterval time.Duration) (*LrOSCustomHeaderPost202Retry200Response, error) {
 	return nil, nil
 }
 
+// Response returns the last HTTP response.
+func (p *lrOSCustomHeaderPost202Retry200Poller) response() *azcore.Response {
+	if p.pt == nil {
+		return nil
+	}
+	return p.pt.latestResponse()
+}
+
+// doneWithContext queries the service to see if the operation has completed.
+func (p *lrOSCustomHeaderPost202Retry200Poller) doneWithContext(ctx context.Context) (done bool, err error) {
```

Comment:
Rename these methods to `done(ctx context.Context)` (the `WithContext` was due to pre-Go 1.7 implementation when `context.Context` didn't yet exist).

Reply:
Done